NuRadioReco/modules/efieldToVoltageConverter.py

```diff
@@ -187,10 +185,25 @@ def run(self, evt, station, det):
                         time_remainder = start_time - start_bin * time_resolution
                     self.logger.debug('channel {}, start time {:.1f} = bin {:d}, ray solution {}'.format(channel_id, electric_field.get_trace_start_time() + cab_delay, start_bin, electric_field[efp.ray_path_type]))
                     new_efield.apply_time_shift(time_remainder)
-                    new_trace[:, start_bin:(start_bin + new_efield.get_number_of_samples())] = new_efield.get_trace()
+
+                    tr = new_efield.get_trace()
+                    stop_bin = start_bin + new_efield.get_number_of_samples()
+
+                    # if checks should never be true...
+                    if stop_bin > np.shape(new_trace)[-1]:
+                        # ensure new efield does not extend beyond end of trace although this should not happen
+                        self.logger.warning("electric field trace extends beyond the end of the trace and will be cut.")
+                        stop_bin = np.shape(new_trace)[-1]
+                        tr = np.atleast_2d(tr)[:,:stop_bin-start_bin]
+                    if start_bin < 0:
+                        # ensure new efield does not extend beyond start of trace although this should not happen
+                        self.logger.warning("electric field trace extends beyond the beginning of the trace and will be cut.")
+                        tr = np.atleast_2d(tr)[:,-start_bin:]
+                        start_bin = 0
+                    new_trace[:, start_bin:stop_bin] = tr
                 trace_object = NuRadioReco.framework.base_trace.BaseTrace()
                 trace_object.set_trace(new_trace, 1. / time_resolution)
-                trace_object.set_trace_start_time(np.min(times_min) - cab_delay)
+                trace_object.set_trace_start_time(times_min.min() - cab_delay)
```

Comment:
@cg-laser (private comm): "should be correct. cable delay is to be considered in forward simulation, since digitizer sees the pulse indeed later than it actually arrives."

Reply:
true, this line is not needed anywhere